sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala

```diff
@@ -180,7 +180,8 @@ class HiveQuerySuite extends HiveComparisonTest with SQLTestUtils with BeforeAnd
     "SELECT CAST(CAST('NaN' AS DOUBLE) AS DECIMAL(1,1)) FROM src LIMIT 1")
 
   createQueryTest("constant null testing",
-    """SELECT
+    """set spark.sql.legacy.allowCastNumericToTimestamp=true;
```

Comment:
We can update `CAST(1 AS TIMESTAMP)` to `CAST('1970-01-01 10:10:10' AS TIMESTAMP)`